RxSwift/Observables/Merge.swift

```diff
@@ -170,7 +170,10 @@ private final class MergeLimitedSinkIter<SourceElement, SourceSequence: Observab
         case .completed:
             self.parent.group.remove(for: self.disposeKey)
             if let next = self.parent.queue.dequeue() {
-                self.parent.subscribe(next, group: self.parent.group)
+                _ = CurrentThreadScheduler.instance.schedule(()) { _ in
+                    self.parent.subscribe(next, group: self.parent.group)
```

Comment:
yea, I think you are right that there is an issue and I need to ensure the `NSRecursiveLock` is re-locking inside this block. I think for the current locking cycle from `synchronized_on`, there is no more mutation after this line to worry about. The lock will recursively lock on the next inner subscribe.

There is a separate issue here which is that the the disposable from ` _ = CurrentThreadScheduler.instance.schedule(())` needs to be added to the `group` otherwise this will not dispose if the `CompositeDisposable` is disposed. This is minor though since the `parent.subscribe` would have no impact.